Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -480,40 +536,51 @@ namespace AZ::DocumentPropertyEditor
                 {
                     hashValue = true;
                 }
-                VisitValue(
-                    instancePointerValue,
-                    access.Get(),
-                    attributes,
-                    // this needs to write the value back into the reflected object via Json serialization
-                    [valuePointer = access.Get(), valueType = access.GetType(), this](const Dom::Value& newValue)
-                    {
-                        // marshal this new value into a pointer for use by the Json serializer
-                        auto marshalledPointer = AZ::Dom::Utils::TryMarshalValueToPointer(newValue, valueType);
-
-                        rapidjson::Document buffer;
-                        JsonSerializerSettings serializeSettings;
-                        JsonDeserializerSettings deserializeSettings;
-                        serializeSettings.m_serializeContext = m_serializeContext;
-                        deserializeSettings.m_serializeContext = m_serializeContext;
-
-                        // serialize the new value to Json, using the original valuePointer as a reference object to generate a minimal diff
-                        JsonSerialization::Store(buffer, buffer.GetAllocator(), marshalledPointer, valuePointer, valueType, serializeSettings);
-
-                        // now deserialize that value into the original location
-                        JsonSerialization::Load(valuePointer, valueType, buffer, deserializeSettings);
-
-                        // NB: the returned value for serialized pointer values is instancePointerValue, but since this is passed by pointer,
-                        // it will not actually detect a changed dom value. Since we are already writing directly to the DOM before this step,
-                        // it won't affect the calling DPE, however, other DPEs pointed at the same adapter would be unaware of the change,
-                        // and wouldn't update their UI.
-                        // In future, to properly support multiple DPEs on one adapter, we will need to solve this. One way would be to store
-                        // the json serialized value (which is mostly human-readable text) as an attribute, so any change to the Json would
-                        // trigger an update. This would have the advantage of allowing opaque and pointer types to be searchable by the
-                        // string-based Filter adapter. Without this, things like Vector3 will not have searchable values by text. These
-                        // advantages would have to be measured against the size changes in the DOM and the time taken to populate and parse them.
-                        return newValue;
-                    },
-                    false, hashValue);
+
+                if (IsInspectorOverrideManagementEnabled() && !serializedPathAttribute.GetString().empty())
```

Comment:
I'd rather not have IsInspector-type stuff here. Let's find a way to hook into this that isn't Inspector specific, something like your OnBeginRow mechanism above. You can call a virtual function here that you override in your derived adapter, or something like that. There are editors (like the AssetEditor) that use the ReflectionAdapter but have nothing to do with prefabs or overrides. Let's keep this clean.

Reply:
Similar to my other response, the IsInspectorOverrideManagementEnabled() function is simply a feature flag check and is only temporary and will go away.

The serialized path attribute mechanism is a new way of storing opaque types in DPE dom, where we will be storing fully serialized values( if there is a serialized path associated with the value). It's not prefab or inspector specific code. So it belongs in this class imo and will become the new way of storing opaque values in DPE DOM after the work in https://github.com/o3de/sig-content/issues/112 finishes. 

I just put this behind a flag so that we can test it properly and not impact other DPE editor already released like the cvar editor.